scispacy/abbreviation.py

```diff
@@ -61,17 +61,17 @@ def find_abbreviation(
         long_index -= 1
         short_index -= 1
 
-    # If we complete the string, we end up with -1 here,
-    # but really we want all of the text.
-    long_index = max(long_index, 0)
+    # The last subtraction takes us on to a space character
+    long_index += 1
 
     # Now we know the character index of the start of the character span,
     # here we just translate that to the first token beginning after that
     # value, so we can return a spaCy span instead.
     word_lengths = 0
     starting_index = None
     for i, word in enumerate(long_form_candidate):
-        word_lengths += len(word)
+        # need to add 1 for the space characters
```

Comment:
Hmm, I'm not sure about this. Is it guaranteed that spacy represents token boundaries with a single whitespace, rather than multiple? Because here you assume it's always one. I think this was working before because we rarely encounter a long form candidate which contains single letter words, and then it would still work if we are off by one. 

I think the right thing to do here might be to use the `text_with_ws` attribute from spacy when we construct the long form, so that we match their whitespace exactly.

Reply:
I think you're right that `text_with_ws` is the right thing to use, but it was more than an off by one thing before, because the `long_form_candidate` could contain multiple spaces. Before there was some lucky math combining the off by one from above and the number of words we take before the abbreviation. For example:

```
In [24]: text = "this text has a bunch more words this text has five words before the Newsss Abbreviation Text Under All (NATURAL)"

In [25]: nlp(text)._.abbreviations[0]._.long_form
Out[25]: Abbreviation Text Under All
```

Reply:
Ok, updated to use `text_with_ws` and added a test that has multiple white spaces in the string.